### PR TITLE
esp32,rp2: Support common BUILD_VERBOSE and V=1 build flags.

### PR DESCRIPTION
**Pull request by projectgus**
### Summary

Small build system consistency/quality-of-life improvement:

* Adds the same `V=1` or `BUILD_VERBOSE` flags that exist on other ports to the esp32 and rp2 ports.
* Previously, to get verbose build output had to run rp2 build with `VERBOSE=1` (undocumented) and esp32 with the same (for Makefile generator) or by running `idf.py` manually (for Ninja generator).

### CODE REVIEW

File: ports/rp2/Makefile [hunk 1]
```diff
@@ -28,7 +30,9 @@ else
 BUILD ?= build-$(BOARD)
 endif
 
-$(VERBOSE)MAKESILENT = -s
```
Comment (projectgus) [inline ports/rp2/Makefile:31]: This PR no longer passes `-s` to the top-level make invocation, but this appears not necessary as the generated Makefile sets `.SILENT` at the top level based on the `VERBOSE` variable. I couldn't see a difference with or without `-s` at the top level.

File: ports/rp2/Makefile [hunk 2]
```diff
@@ -28,7 +30,9 @@ else
...
+ifeq ($(BUILD_VERBOSE),1)
+    MAKE_ARGS += VERBOSE=1  # Picked up in Makefile generated by CMake
```
Comment (dpgeorge) [inline ports/rp2/Makefile:34]: Maybe un-indent this, to match the ifdef's below?

**Comment by dpgeorge** [comment]
This looks good to me.

File: ports/rp2/Makefile (hunk 2 above)
Comment (projectgus) [reply ports/rp2/Makefile]: Oops, yes thanks. Extra indenting in Makefiles is a dangerous game :laughing:

**Review by dpgeorge — APPROVED** [review]

**Pull request merged**